components/mediation-ntask/src/main/java/org/wso2/carbon/mediation/ntask/NTaskTaskManager.java

```diff
@@ -533,88 +548,88 @@ private String managerId() {
-			} catch (Exception e) {
-				logger.error("Cannot return task status [" + taskName + "]. Error: " +
-				                     e.getLocalizedMessage(), e);
-			}
-		}
-		return false;
-	}
+    @Override
+    public boolean isTaskBlocked(String taskName) {
+        if (!isInitialized()) {
+            return false;
+        }
+        synchronized (lock) {
+            if (taskManager == null) {
+                logger.warn("#isTaskRunning Could not determine the state of the task [" +
+                            taskName + "]. Task manager is not available.");
+                return false;
+            }
+            try {
+                return taskManager.getTaskState(taskName)
+                                  .equals(org.wso2.carbon.ntask.core.TaskManager.TaskState.BLOCKED);
+            } catch (Exception e) {
+                logger.error("Cannot return task status [" + taskName + "]. Error: " +
+                                     e.getLocalizedMessage(), e);
+            }
+        }
+        return false;
+    }
 
-	@Override
-	public boolean isTaskRunning(String taskName) {
-		if (!isInitialized()) {
-			return false;
-		}
-		synchronized (lock) {
-			if (taskManager == null) {
-				logger.warn("#isTaskRunning Could not determine the state of the task [" +
-				            taskName + "]. Task manager is not available.");
-				return false;
-			}
-			try {
-				return taskManager.getTaskState(taskName)
-				                  .equals(org.wso2.carbon.ntask.core.TaskManager.TaskState.NORMAL);
-			} catch (Exception e) {
-				logger.error("Cannot return task status [" + taskName + "]. Error: " +
-				                     e.getLocalizedMessage(), e);
-			}
-		}
-		return false;
-	}
+    @Override
+    public boolean isTaskRunning(String taskName) {
+        if (!isInitialized()) {
+            return false;
+        }
+        synchronized (lock) {
+            if (taskManager == null) {
+                logger.warn("#isTaskRunning Could not determine the state of the task [" +
+                            taskName + "]. Task manager is not available.");
+                return false;
+            }
+            try {
+                return taskManager.getTaskState(taskName)
+                                  .equals(org.wso2.carbon.ntask.core.TaskManager.TaskState.NORMAL);
+            } catch (Exception e) {
+                logger.error("Cannot return task status [" + taskName + "]. Error: " +
+                                     e.getLocalizedMessage(), e);
+            }
+        }
+        return false;
+    }
 
-	private void updateAndCleanupObservers() {
-		Iterator<TaskManagerObserver> i = observers.iterator();
-		while (i.hasNext()) {
-			TaskManagerObserver observer = i.next();
-			observer.update();
-			i.remove();
-		}
-	}
+    private void updateAndCleanupObservers() {
+        Iterator<TaskManagerObserver> i = observers.iterator();
```

Comment:
This was fixed in pull request https://github.com/wso2/carbon-mediation/pull/258